libyul/optimiser/VarNameCleaner.cpp

```diff
@@ -108,7 +110,11 @@ YulString VarNameCleaner::findCleanName(YulString const& _name) const
 
 bool VarNameCleaner::isUsedName(YulString const& _name) const
 {
-	return m_dialect.builtin(_name) || m_usedNames.count(_name);
+	if (m_dialect.builtin(_name) || m_usedNames.count(_name))
```

Comment:
Is `_name` allowed to be empty or do we also need `_name.empty()`? If so, can this be shared with `NameDispenser::illegalName()`?

Reply:
I think it's difficult to share the code there, but I can move the empty check here.

Reply:
Also note that the EVMDialect stuff below will be removed once we can move the EVM instructions to proper builtins.